Kitodo-Query-URL-Import/src/main/java/org/kitodo/queryurlimport/QueryURLImport.java

```diff
@@ -300,6 +298,32 @@ private DataRecord performQueryToRecord(String queryURL, String identifier) thro
         }
     }
 
+    /**
+     * Encodes a parameter value as part of the query of a URL. Characters
+     * outside of the readable ASCII range are converted into hexadecimal
+     * representations, preceded by a percent sign (so-called “percent coding”).
+     *
+     * @param value
+     *            value to encode
+     * @param charset
+     *            charset to use for encoding
+     * @return encoded value
+     */
+    /*
+     * The exception announced by {@code URLEncoder} cannot occur if a Charset
+     * object is used.
+     *
+     * TODO: This method can be replaced by URLEncoder.encode(String, Charset)
+     * in Java ≥ 10.
+     */
+    private static String encodeQueryParameter(String value, Charset charset) {
+        try {
+            return URLEncoder.encode(value, charset.toString());
+        } catch (UnsupportedEncodingException e) {
```

Comment:
why are you changing the Exception here?

Reply:
Because `UnsupportedEncodingException` is a checked exception which you would have to declare everywhere, but it cannot be thrown since the method takes a `Charset`, not an arbitrary string.
Declaring an exception that is never thrown is bad design.